Source/Scene/Model.js

```diff
@@ -4249,28 +4050,34 @@ define([
             // Textures may continue to stream in while in the LOADED state.
             if (loadResources.pendingBufferLoads === 0) {
                 if (!loadResources.initialized) {
-                    // glTF pipeline updates
-                    var options = {
-                        optimizeForCesium: true,
-                        addBatchIdToGeneratedShaders : this._addBatchIdToGeneratedShaders
-                    };
                     frameState.brdfLutGenerator.update(frameState);
-                    updateVersion(this.gltf);
-                    if (defined(this.gltf.asset) && defined(this.gltf.asset.extras) &&
-                        this.gltf.asset.extras.gltf_pipeline_upgrade_10to20) {
-                            this._forwardAxis = Axis.X;
-                    }
+
                     ModelUtility.checkSupportedExtensions(this.extensionsRequired);
-                    addPipelineExtras(this.gltf);
-                    addDefaults(this.gltf);
-                    processModelMaterialsCommon(this.gltf, options);
-                    processPbrMetallicRoughness(this.gltf, options);
+                    ModelUtility.updateForwardAxis(this);
+
+                    // glTF pipeline updates, not needed if loading from cache
+                    if (!this._loadRendererResourcesFromCache) {
+                        var gltf = this.gltf;
+                        // Add the original version so it remains cached
+                        gltf.extras.sourceVersion = gltf.asset.version;
```

Comment:
I ran into an old tileset that doesn't actually set `gltf.asset` which is technically valid for gltf 1.0. The code here and in `ModelUtility` should account for that and assume `1.0` if asset is undefined.